src/UserCredentials.cs

```diff
@@ -35,6 +52,20 @@ public UserCredentials(string username, string password)
             _password = password;
         }
 
+        /// <summary>
+        /// Creates a <see cref="UserCredentials"/> class based on a domain and username that does not require a password.
+        /// </summary>
+        /// <param name="domain">The domain.</param>
+        /// <param name="username">The username.</param>
+        public UserCredentials(string domain, string username)
```

Comment:
This constructor takes two string params, which is already being used by the constructor before it - so it will error at compile time.

How about using static factory methods instead?  `UserCredentials.CreateWithoutPassword(domain, username)`.  Same for the other one.